### PR TITLE
8341298: Open source more AWT window tests

### PR DESCRIPTION
**Pull request by jayathirthrao**
Clean up and open source more of the AWT window tests.
Automated tests `PopupProxyCrash` and `NoResizeEvent` are also verified in our CI.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341298](https://bugs.openjdk.org/browse/JDK-8341298): Open source more AWT window tests (**Bug** - P4)

### Reviewers
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21348/head:pull/21348` \
`$ git checkout pull/21348`

Update a local copy of the PR: \
`$ git checkout pull/21348` \
`$ git pull https://git.openjdk.org/jdk.git pull/21348/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21348`

View PR using the GUI difftool: \
`$ git pr show -t 21348`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21348.diff">https://git.openjdk.org/jdk/pull/21348.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21348#issuecomment-2393035197)

### CODE REVIEW

File: test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+import java.awt.event.ItemEvent;
+import java.awt.event.ItemListener;
+import java.util.Vector;
+
+public class TestLocationByPlatformWithControls extends Frame
+    implements ActionListener, ItemListener {
+    Panel northP, centerP;
+    Checkbox undecoratedCB, defaultLocationCB, visibleCB,
+        iconifiedCB, maximizedCB;
+    Button createB, packB, moveB, resizeB, reshapeB, disposeB;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java:50]: It is recommended to declare variables on separate lines.

File: test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+    Vector frames;
+    public static void main(String[] args) throws Exception {
+        String INSTRUCTIONS = """
+            This test is to check that LocationByPlatform works with  other
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java:54]:
```suggestion
            This test is to check that LocationByPlatform works with other
```

File: test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+            Frame frame = new Frame();
+            frame.setSize(200, 200);
+            frames.add(frame);
+            updateControls(frame);
+            Panel panel = new Panel();
+            frame.add(panel);
+            panel.add(new Button ("Test Button"));
+            panel.add(new Button ("Test Button 1"));
+            panel.add(new Button ("Test Button  2"));
+            panel.add(new Button ("Test Button   3"));
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java:158]:
```suggestion
            panel.add(new Button ("Test Button 2"));
            panel.add(new Button ("Test Button 3"));
```

File: test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+import javax.swing.Box;
+import javax.swing.JComboBox;
+import javax.swing.JTextField;
+import javax.swing.plaf.basic.BasicComboBoxUI;
+import javax.swing.plaf.basic.BasicComboPopup;
+import javax.swing.plaf.basic.ComboPopup;
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Color;
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java:41]: Awt imports before swing.

File: test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    }
+
+    static void runTest() throws Exception {
+        for (int i = 0; i < TEST_COUNT; i++) {
+            Point point = getComboBoxLocation();
+            robot.mouseMove(point.x, point.y);
+            robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+            robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+            robot.delay(500);
+            robot.waitForIdle();
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java:167]: Seen many review comments where it is advised to invoke delay after waitForIdle.

File: test/jdk/java/awt/Window/WindowToFrontTest/WindowToFrontTest.java [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+        PassFailJFrame passFailJFrame = PassFailJFrame.builder()
+            .title("Test Instructions")
+            .instructions(INSTRUCTIONS)
+            .rows(13)
+            .columns(40)
+            .build();
+        EventQueue.invokeAndWait(() -> createUI());
+        passFailJFrame.awaitAndCheck();
+    }
```
Comment (kumarabhi006) [inline test/jdk/java/awt/Window/WindowToFrontTest/WindowToFrontTest.java:62]: Please add a blank line after this.

File: test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java (hunk 1 above)
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java]: Updated.

File: test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java (hunk 2 above)
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/LocationByPlatformWithControls/TestLocationByPlatformWithControls.java]: Updated.

File: test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+        Container parent = Box.createVerticalBox();
+        parent.add(jtf);
+        parent.add(tf);
+        parent.add(panel);
+        parent.add(cb);
+
+        frame.add(parent, BorderLayout.CENTER);
+        frame.setSize(200,200);
+        frame.pack();
```
[the remark this answers is not in the document]
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java]: Updated.

File: test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java (hunk 4 above)
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java]: Updated.

File: test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java (hunk 5 above)
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/ProxyCrash/PopupProxyCrash.java]: Updated.

File: test/jdk/java/awt/Window/WindowToFrontTest/WindowToFrontTest.java (hunk 6 above)
Comment (jayathirthrao) [reply test/jdk/java/awt/Window/WindowToFrontTest/WindowToFrontTest.java:62]: Updated.

**Review by kumarabhi006 — APPROVED** [review]

**Comment by jayathirthrao** [comment]
Thanks for the review.
Updated automated tests are again verified in our CI.

**Pull request closed without merging**